### PR TITLE
[OpenTracing] Add skeleton code

### PR DESCRIPTION
**Pull request by delner**
This pull request extends OpenTracing components in their most basic form, which are all no-ops. It should act as a starting point for adding code and tests as we begin to implement OpenTracing for `ddtrace`.

### CODE REVIEW

File: spec/ddtrace/opentracing/span_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+require 'ddtrace/opentracing'
+require 'ddtrace/opentracing/helper'
+
+if Datadog::OpenTracing.supported?
```
Comment (pawelchcki) [inline spec/ddtrace/opentracing/span_spec.rb]: Could this be moved into RSpec filter?

File: lib/ddtrace/opentracing.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module Datadog
+  # Namespace for ddtrace OpenTracing implementation
+  module OpenTracing
```
Comment (Kyle-Verhoog) [inline lib/ddtrace/opentracing.rb]: In both our python and go opentracing implementations we refer to the tracer and name the sub-module as 'opentracer'.

This is probably not a big deal, but we may wish to be consistent across our implementations.

File: lib/ddtrace/opentracing/carrier.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module Datadog
+  module OpenTracing
+    class Carrier < ::OpenTracing::Carrier
```
Comment (Kyle-Verhoog) [inline lib/ddtrace/opentracing/carrier.rb]: 👍 the python opentracing library does not define a class for the carrier and is assumed to be a `dict`, this is better.

File: spec/ddtrace/opentracing/carrier_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+
+require 'ddtrace/opentracing'
+require 'ddtrace/opentracing/helper'
+
+if Datadog::OpenTracing.supported?
```
Comment (Kyle-Verhoog) [inline spec/ddtrace/opentracing/carrier_spec.rb]: 👍

File: lib/ddtrace/opentracing.rb (hunk 2 above)
Comment (delner) [reply lib/ddtrace/opentracing.rb]: I see. For consistency with all other Datadog tracers, I'll change `OpenTracing` to `OpenTracer`.

**Pull request merged**